model/websocket_client.go

```diff
@@ -6,24 +6,28 @@ package model
 import (
 	"encoding/json"
 	"net/http"
+	"time"
 
 	"github.com/gorilla/websocket"
 )
 
 const (
 	SOCKET_MAX_MESSAGE_SIZE_KB = 8 * 1024 // 8KB
+	PING_TIMEOUT_BUFFER        = 5
 )
 
 type WebSocketClient struct {
-	Url             string          // The location of the server like "ws://localhost:8065"
-	ApiUrl          string          // The api location of the server like "ws://localhost:8065/api/v3"
-	ConnectUrl      string          // The websocket URL to connect to like "ws://localhost:8065/api/v3/path/to/websocket"
-	Conn            *websocket.Conn // The WebSocket connection
-	AuthToken       string          // The token used to open the WebSocket
-	Sequence        int64           // The ever-incrementing sequence attached to each WebSocket action
-	EventChannel    chan *WebSocketEvent
-	ResponseChannel chan *WebSocketResponse
-	ListenError     *AppError
+	Url                string          // The location of the server like "ws://localhost:8065"
+	ApiUrl             string          // The api location of the server like "ws://localhost:8065/api/v3"
+	ConnectUrl         string          // The websocket URL to connect to like "ws://localhost:8065/api/v3/path/to/websocket"
+	Conn               *websocket.Conn // The WebSocket connection
+	AuthToken          string          // The token used to open the WebSocket
+	Sequence           int64           // The ever-incrementing sequence attached to each WebSocket action
+	PingTimeoutChannel chan bool       // The channel used to signal ping timeouts
+	EventChannel       chan *WebSocketEvent
+	ResponseChannel    chan *WebSocketResponse
+	ListenError        *AppError
+	pingTimeoutTimer   *time.Timer
```

Comment:
this dont need to be starting with capital letter? `PingTimeoutTimer`

Reply:
The timer is not exposed but is only used internally in the struct, so it's lowercase. Otherwise it'd be exposed and could be modified externally.

Reply:
thanks!